### ISSUE TITLE
Build error: "cannot use makeInsecureTransport"

### ISSUE THREAD

**Issue by RubenKelevra** [libp2p/hydra-booster#120]
I tried to build hydra-booster via:

```
$ go get -u github.com/libp2p/hydra-booster
```

And Go 1.16.

I get this error:

```
$ go get -u github.com/libp2p/hydra-booster
# github.com/libp2p/go-libp2p/config
go/pkg/mod/github.com/libp2p/go-libp2p@v0.13.0/config/config.go:144:19: cannot use makeInsecureTransport(h.ID(), cfg.PeerKey) (type sec.SecureTransport) as type sec.SecureMuxer in assignment:
        sec.SecureTransport does not implement sec.SecureMuxer (wrong type for SecureInbound method)
                have SecureInbound(context.Context, net.Conn) (sec.SecureConn, error)
                want SecureInbound(context.Context, net.Conn) (sec.SecureConn, bool, error)
go/pkg/mod/github.com/libp2p/go-libp2p@v0.13.0/config/config.go:146:24: cannot assign sec.SecureTransport to upgrader.Secure (type sec.SecureMuxer) in multiple assignment:
        sec.SecureTransport does not implement sec.SecureMuxer (wrong type for SecureInbound method)
                have SecureInbound(context.Context, net.Conn) (sec.SecureConn, error)
                want SecureInbound(context.Context, net.Conn) (sec.SecureConn, bool, error)
go/pkg/mod/github.com/libp2p/go-libp2p@v0.13.0/config/security.go:56:2: cannot use secMuxer (type *csms.SSMuxer) as type sec.SecureTransport in return argument:
        *csms.SSMuxer does not implement sec.SecureTransport (wrong type for SecureInbound method)
                have SecureInbound(context.Context, net.Conn) (sec.SecureConn, bool, error)
                want SecureInbound(context.Context, net.Conn) (sec.SecureConn, error)
go/pkg/mod/github.com/libp2p/go-libp2p@v0.13.0/config/security.go:78:2: cannot use secMuxer (type *csms.SSMuxer) as type sec.SecureTransport in return argument:
        *csms.SSMuxer does not implement sec.SecureTransport (wrong type for SecureInbound method)
                have SecureInbound(context.Context, net.Conn) (sec.SecureConn, bool, error)
                want SecureInbound(context.Context, net.Conn) (sec.SecureConn, error)
```

**Comment by aschmahmann** [comment]
TLDR `go get -u` just flat out doesn't work on go modules that are pre v2 (which generally makes it not helpful if any of your dependencies are pre v2). `go get -u=patch` will generally work unless someone's name a versioning mistake. 

What's happening here is you are auto upgrading dependencies with breaking changes (pre v1 minor versions bumps are considered breaking), that's a bad idea and will likely cause you compile errors.

I'm closing this issue, but if you want to file an issue or PR to update the dependencies you'd like updated (e.g. go-libp2p) then that seems fine by me.

**Comment by RubenKelevra** [comment]
@aschmahmann I'm not sure I can follow. I just cloned this repo and copied the command mentioned under "install" in the readme.

**Comment by aschmahmann** [comment]
Well, that seems like a bug in the README. If you clone the repo you can do `go build` in the root directory. Go also has some tools to let you install binaries without cloning the repo (e.g. via `go install {repourl}@{version}`), but none of that is specific to this repo just the Go binary.
